### ISSUE TITLE
Add Urdf writers

### ISSUE THREAD

**Issue by gonzalocasas** [compas-dev/compas#670]
Currently, the URDF support can only read/parse urdf files, but cannot write them back to disk if they are modified.
Adding urdf writer support would help in other areas, such as mix&match different robot clients in compas_fab

**Comment by gonzalocasas** [comment]
See https://github.com/compas-dev/compas_fab/issues/232 for more details

**Comment by beverlylytle** [comment]
Not losing information from a urdf which is read into the system is important, but with the xml parser we are using not entirely possible, no "comments, processing instructions, and document type declarations." See the note at the bottom of the section [here](https://docs.python.org/3/library/xml.etree.elementtree.html#parsing-xml).

**Comment by tomvanmele** [comment]
could we upgrade our parser with something like this?
https://dustinoprea.com/2019/01/22/python-parsing-xml-and-retaining-the-comments/

**Comment by brgcode** [comment]
or use BeautifulSoup instead?

**Comment by beverlylytle** [comment]
> could we upgrade our parser with something like this?

I looked into this, but I was unsuccessful.  There were a few obstructions.  One is that many URDFs have comments outside the root element of the xml, which, pre python 3.8, xml.etree strongly objects to.  When I tried to snake my way around that, I discovered that there were many changes made within xml.etree over the various versions of python 3, and it just got too complicated.
